dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/RegexUtils.java

```diff
@@ -36,4 +40,10 @@ private RegexUtils(){}
      */
     public static final String REGEX_REQUESTMAPPING_IDENTIFIER_AS_DIGIT = "/{id:\\d+}";
 
+    /**
+     * Regular expression to accept a string of 32 hexadecimal digits.
+     */
+    public static final String REGEX_REQUESTMAPPING_IDENTIFIER_AS_HEX32
+            = "/{id:[0-9a-zA-Z]{32}}";
```

Comment:
This looks to be matching *too much*.  Aren't any alphabetic letters past F invalid in hexidecimal?  I think this should be more like
`/{id:[0-9a-fA-F]{32}`

I should also point out here that it looks like it _could be possible_ for a very large, 32-digit number to match both `REGEX_REQUESTMAPPING_IDENTIFIER_AS_DIGIT` and `REGEX_REQUESTMAPPING_IDENTIFIER_AS_HEX32`.  But, I'm not sure how to work around that, and the likelihood seems very slim that would ever occur (as numerical identifiers are sequential in DSpace)

Reply:
Right, thank you for catching that.  Indeed we only need a-fA-F.

32 hex digits (16 bytes) is much larger than a Java `long`.  It could only be held in a BigInteger.  I studied the problem for some time before deciding that I could do no better than this.